src/errors/index.js

```diff
@@ -1,5 +1,7 @@
 'use strict'
 
+// TODO: Should be refactored when https://github.com/ipfs/js-ipfs/pull/1746/ is adopted.
```

Comment:
In the PR there are some utils to ease the creation of the Error's class, but since I believe it will be moved out to some general package to be used across `js` land, I did not copy&pasted it directly here and just created the Error's classes manually. Once it will land this can be adopted here.